### PR TITLE
Created competency model pages

### PR DESCRIPTION
**Pull request by 0xJericLloyd**
1. Created CoreModels, CrossCuttingModels, FunctionalModels, and ManagerialModels.js page and its components:

- AddCompetencyModelModal.js
- DeleteCompetencyModelModal.js
- EditCompetencyModelModal.js

2. Created TableSalaryGrade.js component

3. Created Competency Model store

### CODE REVIEW

File: src/components/Modal/Competency/AddCompetencyModelModal.js [hunk 1]
```diff
@@ -61,28 +81,29 @@ const AddCompetencyModelModal = props => {
...
       ),
     }),
     onSubmit: (values, { resetForm }) => {
-      console.log(values)
-      //   dispatch(addCompetencyDetails(values))
-      //   resetForm()
+      const domainId = _id
+      const competencyModelDataSubmit = { ...values, domainId }
+      dispatch(addCompetencyDetails(competencyModelDataSubmit))
+      resetForm()
```
Comment (allyncubero) [inline src/components/Modal/Competency/AddCompetencyModelModal.js:106]: Transfer resetForm to checker if the transaction is successful. 

On submission of the form and the transaction did not go through, the user will have to encode every detail in the form again.

File: src/components/InRowAction/InRowAction.js [hunk 2]
```diff
@@ -68,6 +69,15 @@ const InRowAction = props => {
             <i className="fas fa-edit"></i>
           </button>
         ) : null}
+
```
Comment (allyncubero) [inline src/components/InRowAction/InRowAction.js:72]: Why add this? What's the difference between deleteCompetencyModel and deleteModal? editCompetencyModel and editModal?

Comment (0xJericLloyd) [reply src/components/InRowAction/InRowAction.js]: Apparently, I have not used deleteModal and editModal, which lead to creating another prop/s in InRowAction.

File: src/components/Modal/Competency/AddCompetencyModelModal.js (hunk 1 above)
Comment (0xJericLloyd) [reply src/components/Modal/Competency/AddCompetencyModelModal.js]: Transferred resetForm already to the checker if submission is successful.

**Review by allyncubero — CHANGES_REQUESTED** [review]
Please check comments. Thanks!

**Review by allyncubero — APPROVED** [review]
Good2x

**Pull request merged**